clippy_lints/src/indexing_slicing.rs

```diff
@@ -141,12 +155,18 @@ impl<'tcx> LateLintPass<'tcx> for IndexingSlicing {
                     (None, None) => return, // [..] is ok.
                 };
 
-                span_lint_and_help(cx, INDEXING_SLICING, expr.span, "slicing may panic", None, help_msg);
+                span_lint_and_then(cx, INDEXING_SLICING, expr.span, "slicing may panic", |diag| {
+                    diag.help(help_msg);
+
+                    if cx.tcx.hir().is_inside_const_context(expr.hir_id) {
+                        diag.note(note);
+                    }
+                });
             } else {
                 // Catchall non-range index, i.e., [n] or [n << m]
                 if let ty::Array(..) = ty.kind() {
                     // Index is a const block.
-                    if let ExprKind::ConstBlock(..) = index.kind {
```

Comment:
After reading the `.stderr` output again, I understand this check better. This is exclusive for arrays. Arrays can be bounds checked during compile time, which makes indexing with const index values safe again.

This currently triggers the uitest in line 28 and 29 which it shouldn't. Could you remove the `self.suppress_restriction_lint_in_const` check again from this if?